framework/src/auxkernels/ProjectionAux.C

```diff
@@ -0,0 +1,113 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ProjectionAux.h"
+#include "SystemBase.h"
+
+registerMooseObjectRenamed("MooseApp", SelfAux, "01/30/2024 24:00", ProjectionAux);
+registerMooseObject("MooseApp", ProjectionAux);
+
+InputParameters
+ProjectionAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.addClassDescription(
+      "Returns the specified variable as an auxiliary variable with a projection of the source "
+      "variable. If they are the same type, this amounts to a simple copy.");
+  params.addRequiredCoupledVar("v", "Variable to take the value of.");
+
+  // Technically possible to project from nodal to elemental and back
+  params.set<bool>("_allow_nodal_to_elemental_coupling") = true;
+
+  // We need some ghosting for all elemental to nodal projections
+  params.addParam<unsigned short>("ghost_layers", 1, "The number of layers of elements to ghost.");
+  params.addRelationshipManager(
+      "ElementPointNeighborLayers",
+      Moose::RelationshipManagerType::GEOMETRIC | Moose::RelationshipManagerType::ALGEBRAIC,
```

Comment:
changed to algebraic. I thought it made sense to have geometric when you need element info and algebraic as well for variable info

Reply:
It does make sense!  So much sense that we don't ever do the latter without the former.

Reply:
Maybe I shouldn't have unsubscribed. We don't add MOOSE algebraic ghosting functors to the mesh, so the `|` is needed

Reply:
So this works because of the default geometric ghosting in libMesh (and also because of Exodus output turned on in this input file). If I disable the default geometric ghosting in libMesh (which we never do in MOOSE so this is somewhat of a hypothetical exercise) and turn off exodus output, then I get some exodiffs with distributed mesh

Reply:
I don't understand why the heck I added `MooseGhostPointNeighborLayers` when we have `ElementPointNeighborLayers`. Opened https://github.com/idaholab/moose/issues/24573 to remove it

Reply:
>We don't add MOOSE algebraic ghosting functors to the mesh

And ... of course we don't.  `/*to_mesh = */ false`.  I'd forgotten that I'd given up on fighting the `break_encapsulation_because_thats_easier_to_optimize` option; sorry for both stages of that.